src/Client.php

```diff
@@ -371,7 +371,19 @@ public function uploadChunked(string $path, $contents, $mode = 'add', $chunkSize
             $chunkSize = $this->maxChunkSize;
         }
 
-        $stream = Psr7\stream_for($contents);
+        if ($this->isPipe($contents)) {
+            /** @var resource $contents */
+            $stream = new \GuzzleHttp\Psr7\PumpStream(function ($length) use ($contents) {
+                $data = fread($contents, $length);
+                if (strlen($data) === 0) {
+                    return false;
+                }
+
+                return $data;
+            });
+        } else {
```

Comment:
Is this else statement needed?

Reply:
Yes, it's a speedup for non-piped contents...which I believe is the majority. :)

Reply:
But won't the results be the same if the `else` is removed? There's a `return` statement for each path inside the `if` above.

Reply:
This is the whole chunk:
`
        if ($this->isPipe($contents)) {
            /** @var resource $contents */
            $stream = new \GuzzleHttp\Psr7\PumpStream(function ($length) use ($contents) {
                $data = fread($contents, $length);
                if (strlen($data) === 0) {
                    return false;
                }
                return $data;
            });
        } else {
            $stream = Psr7\stream_for($contents);
        }`

Return statement is inside a closure...so it doesn't affect the outer `if`.

The point of these changes is to initialize `$stream` differently depending on whether input stream is piped (in which case it has to be pumped) or not (this is where we can speed it up and use less resources to process it).